spec/tbd_tests_spec.rb

```diff
@@ -3698,4 +3698,278 @@
     expect(vec[2].y).to be_within(TOL).of( 0.25)
     expect(vec[2].z).to be_within(TOL).of(11.83)
   end
+
+  it "can uprate (ALL roof) constructions" do
```

Comment:
As with other recently-added tests, this may revert back to the [TBD Test](https://github.com/rd2/tbd_tests) Suite (once done).